go/client/versionfix.go

```diff
@@ -8,24 +8,73 @@ import (
 	"github.com/blang/semver"
 	"github.com/keybase/client/go/libkb"
 	keybase1 "github.com/keybase/client/go/protocol"
+	rpc "github.com/keybase/go-framed-msgpack-rpc"
 	"golang.org/x/net/context"
+	"io/ioutil"
+	"net"
 	"os"
+	"strconv"
+	"strings"
 	"time"
 )
 
+func getPid(g *libkb.GlobalContext) (int, error) {
+	fn, err := g.Env.GetPidFile()
+	if err != nil {
+		return -1, err
+	}
+	data, err := ioutil.ReadFile(fn)
+	if err != nil {
+		return -1, err
+	}
+	pidString := strings.TrimSpace(string(data))
+	pid, err := strconv.ParseInt(pidString, 10, 64)
+	if err != nil {
+		return -1, err
+	}
+	return int(pid), nil
+}
+
+func killPid(pid int) error {
+	if pid < 0 {
+		return fmt.Errorf("invalid pid given to kill")
+	}
+
+	p, err := os.FindProcess(pid)
+	if err != nil {
+		return err
+	}
+	err = p.Signal(os.Kill)
+	return err
+}
+
 func FixVersionClash(g *libkb.GlobalContext, cl libkb.CommandLine) (err error) {
 	var cli keybase1.ConfigClient
 	var ctlCli keybase1.CtlClient
 	var serviceConfig keybase1.Config
+	var socket net.Conn
 
 	g.Log.Debug("+ FixVersionClash")
 	defer func() {
+		if socket != nil {
+			socket.Close()
```

Comment:
the other place `socket.Close()` happens, it is set to nil right afterwards.  wasn't sure if that should happen after this `socket.Close()` too or not.